store-implementation/parquet-store/src/main/java/uk/gov/gchq/gaffer/parquetstore/operation/addelements/handler/AddElementsHandler.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2017. Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.gchq.gaffer.parquetstore.operation.addelements.handler;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.FileUtil;
+import org.apache.hadoop.fs.Path;
+import org.apache.spark.sql.SparkSession;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.gov.gchq.gaffer.commonutil.iterable.CloseableIterable;
+import uk.gov.gchq.gaffer.commonutil.iterable.CloseableIterator;
+import uk.gov.gchq.gaffer.operation.OperationException;
+import uk.gov.gchq.gaffer.operation.impl.add.AddElements;
+import uk.gov.gchq.gaffer.parquetstore.ParquetStore;
+import uk.gov.gchq.gaffer.parquetstore.operation.addelements.impl.AggregateAndSortTempData;
+import uk.gov.gchq.gaffer.parquetstore.operation.addelements.impl.GenerateIndices;
+import uk.gov.gchq.gaffer.parquetstore.operation.addelements.impl.WriteUnsortedData;
+import uk.gov.gchq.gaffer.spark.SparkUser;
+import uk.gov.gchq.gaffer.store.Context;
+import uk.gov.gchq.gaffer.store.Store;
+import uk.gov.gchq.gaffer.store.StoreException;
+import uk.gov.gchq.gaffer.store.operation.handler.OperationHandler;
+import uk.gov.gchq.gaffer.user.User;
+
+import java.io.IOException;
+import java.util.Iterator;
+
+public class AddElementsHandler implements OperationHandler<AddElements> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AddElementsHandler.class);
+
+    @Override
+    public Void doOperation(final AddElements operation,
+                            final Context context, final Store store)
+            throws OperationException {
+        store.optimiseSchema();
```

Comment:
I don't think this store.optimiseSchema should be here. It will be called automatically for you in the Store class when the store is initialised.